use "application.statusKey" instead of "application.secret" …

asolntsev opened the pull request:
…for /@status requests

theo commented on documentation/commands/cmd-status.txt:
```diff
@@ -25,8 +25,8 @@
 ~
 ~ --secret: 
```
Assuming we're trying to introduce a different key for the purposes of getting status, should we change the flag from `--secret` to `--status-key` to avoid confusion?

asolntsev commented:
Hm... Yes, probably. Good point.

On May 22, 2017 4:35 PM, "Theodore Nguyen-Cao" <notifications@github.com>
wrote:

> *@theo* commented on this pull request.
> ------------------------------
>
> In documentation/commands/cmd-status.txt
> <https://github.com/playframework/play1/pull/1144#discussion_r117746114>:
>
> > @@ -25,8 +25,8 @@
>  ~
>  ~ --secret:
>
> Assuming we're trying to introduce a different key for the purposes of
> getting status, should we change the flag from --secret to --status-key
> to avoid confusion?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/playframework/play1/pull/1144#pullrequestreview-39477100>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARE3bCaCvJTKCGjtnRXLo2L9tcksU3sks5r8Y8ggaJpZM4NiSgT>
> .
>

asolntsev commented:
Thank you @theo ! I renamed `--status` to `--status-key`. 
Additionally, I extracted "/@status" and "/@kill" part of `CorePlugin` to a dedicated class `PlayStatusPlugin` (single responsibility principle). And renamed `CorePlugin` to `EnhancerPlugin` because now it's the only job of this plugin. 

@xael-fry Please also review the changes.

asolntsev commented:
@xael-fry @flybyray Please review this pull request. I would like to merge it.

xael-fry commented:
Merged thanks

sbeigel commented:
How ist this supposed to work? Why did you remove the signing part in the plugin (-> server side)? I thought your goal was to not re-use the app secret but a distinct key. But I think we need to keep the signing to check the message "@status" signed w/ the "new" statusKey?!

xael-fry commented:
@asolntsev I let you answer this question

asolntsev commented:
@sbeigel @xael-fry Sorry for the long replay.
We removed the signing part because it was a potential security problem. It allowed hacker to bruteforce secretKey/statusKey.

sbeigel commented:
But the play status command signs the string "@status" with the application.statusKey property. Then, on the server-side, this signed message is compared to the application.statusKey property. How should this work? This will never be equal :)
To make this work right now, you have to sign the message "@status" by yourself, set this signed message as statusKey in the app.conf and pass *your* signing key as an explicit argument to the play status command... The default case in play status is useless (as it doesn't work).

asolntsev commented:
Sorry, I don't know what means "the default case".

We use the following command to get application status:

curl --header "authorization: YOUR-SECRET-WORD" http://localhost:9000/@status


Andrei Solntsev

2017-11-28 11:13 GMT+02:00 Sebastian Beigel <notifications@github.com>:

> But the play status command signs the string "@status" with the
> application.statusKey property. Then, on the server-side, this signed
> message is compared to the application.statusKey property. How should this
> work? This will never be equal :)
> To make this work right now, you have to sign the message "@status" by
> yourself, set this signed message as statusKey in the app.conf and pass
> *your* signing key as an explicit argument to the play status command...
> The default case in play status is useless (as it doesn't work).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/playframework/play1/pull/1144#issuecomment-347459980>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARE3ZF6CCzegZrP1lRfzN4JeaVY7FxMks5s687QgaJpZM4NiSgT>
> .
>

sbeigel commented:
Yes, that's what I thought :)

But this change effectively renders the play status command useless, as this command (look at the python code) signs the string "@status", either with the given secret or (that is the default) with the app.conf's property statusKey.

I understand that you want a "classic" API/secret key style. But then the play status command has to be converted to this approach as well, i.e. remove the singing part and take the passed argument *or* the app.conf's statusKey as auth message.

asolntsev commented:
Ok, you mean "play status" command?
I never used it.

Yes, sure: if it became broken, it should be fixed.

Anyway, I hope you also agree that it's better to break some insecure
feature than to keep it working.  :)

Andrei Solntsev

2017-11-28 11:25 GMT+02:00 Sebastian Beigel <notifications@github.com>:

> Yes, that's what I thought :)
>
> But this change effectively renders the play status command useless, as
> this command (look at the python code) signs the string "@status", either
> with the given secret or (that is the default) with the app.conf's property
> statusKey.
>
> I understand that you want a "classic" API/secret key style. But then the
> play status command has to be converted to this approach as well, i.e.
> remove the singing part and take the passed argument *or* the app.conf's
> statusKey as auth message.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/playframework/play1/pull/1144#issuecomment-347463067>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARE3YDcv32iIB7id5ar0cLiy-kEzHOKks5s69GIgaJpZM4NiSgT>
> .
>

sbeigel commented:
Yes, I do :)

I can fix the status command (in the way just described) and make a PR if you want...

asolntsev commented:
Sure, it would be great.
Thank you!

Andrei Solntsev

2017-11-28 11:38 GMT+02:00 Sebastian Beigel <notifications@github.com>:

> Yes, I do :)
>
> I can fix the status command (in the way just described) and make a PR if
> you want...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/playframework/play1/pull/1144#issuecomment-347466515>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AARE3SyOKH_cGV-lZaFxN-NWerxl3y3kks5s69SagaJpZM4NiSgT>
> .
>

sbeigel commented:
see #1209
